### ISSUE TITLE
Import resolver prefers typeshed stub over local module with same name

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6406]
This comes from discussion thread https://github.com/microsoft/pyright/discussions/5585#discussioncomment-7525386.

To repro, create the following files within a workspace:
```python
# stripe/__init__.py
```
```python
# stripe/demo.py
from stripe.util import not_in_typeshed # False positive error
```
```python
# stripe/util.py
not_in_typeshed = 0
```

There exists a third-party typeshed module called `stripe/util.pyi`. Currently, pyright is preferring this stub over the local file. It should prefer local modules.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.336, which I just published. It will also be included in a future release of pylance.
